modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to David Pilato (the "Author") under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Author licenses this
+ * file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.testcontainers.elasticsearch;
+
+import org.apache.http.HttpHost;
+import org.testcontainers.containers.GenericContainer;
+
+/**
+ * Represents an elasticsearch docker instance which exposes by default port 9200 and 9300 (transport.tcp.port)
+ * The docker image is by default fetch from docker.elastic.co/elasticsearch/elasticsearch
+ * @author dadoonet
+ */
+public class ElasticsearchContainer extends GenericContainer {
```

Comment:
P.S. don't we need a wait strategy here? AFAIK "listening on a port" is not enough in case of Elasticsearch

Reply:
I thought so as well but apparently this is working OOTB. May be I'm just lucky though. But right, I can implement a similar thing as Couchbase did.